File: storage/data_transpiled/at_coder/arc004/C/3511795.java
public static String cTheMelancholyOfTaroHeikinchi ( String S ) {
  int X = Integer . parseInt ( S ) ;
  int Y = Integer . parseInt ( S ) ;
  ArrayList < int [ ] > ans_tmp = new ArrayList < > ( ) ;
  for ( int N = 2 * X / Y - 1 ;
  N <= 2 * X / Y + 2 ;
  N ++ ) {
    if ( N <= 0 ) continue ;
    int m_numerator = N * ( Y * ( N + 1 ) - 2 * X ) ;
    int m_denominator = 2 * Y ;
    if ( m_numerator % m_denominator != 0 ) continue ;
    int M = m_numerator / m_denominator ;
    if ( 1 <= M && M <= N ) ans_tmp . add ( new int [ ] {
      N , M }
      ) ;
    }
    String ans ;
    if ( ans_tmp . size ( ) > 0 ) {
      ans = "\n" + Arrays . toString ( ans_tmp . toArray ( new int [ ans_tmp . size ( ) ] ) ) ;
    }
    else {
      ans = "Impossible" ;
    }
    return ans ;
  }
  

File: storage/data_original/at_coder/arc004/C/3511795.py
def c_the_melancholy_of_taro_heikinchi(S):
    """
    ?????????
    (1) (1+2+...+N-M)/N=X/Y
    ??????????1+2+...+N=N(N+1)/2 ????????????????
    ????M????????m_numerator, m_denominator???(?????)

    for?????N?????M=1, M=N????N???(1)?????????????
    """
    X, Y = [int(i) for i in S.split('/')]
    ans_tmp = []
    for N in range(2 * X // Y - 1, 2 * X // Y + 2):
        if N <= 0:
            continue

        m_numerator = N * (Y * (N + 1) - 2 * X)
        m_denominator = 2 * Y

        if m_numerator % m_denominator != 0:
            continue  # M ????????

        M = m_numerator // m_denominator
        if 1 <= M <= N:
            ans_tmp.append([N, M])
    if ans_tmp:
        ans = '\n'.join([' '.join(map(str, s)) for s in ans_tmp])
    else:
        ans = 'Impossible'
    return ans

S = input()
print(c_the_melancholy_of_taro_heikinchi(S))